js/freewheel.js

```diff
@@ -385,6 +386,7 @@ OO.Ads.manager(function(_, $) {
      */
     this.playAd = function(ad) {
       _resetAdState();
+      currentAd = ad;
```

Comment:
Slot has getTimePositionClass() which can tell us if it's a preroll, midroll, postroll, or overlay.  Technically it does not differentiate between linear and nonlinear overlays so we assume the overlay to be nonlinear